core/mock/state.go

```diff
@@ -36,133 +36,153 @@ type PrivateOperatorState struct {
+		}
+		if len(oprStakes) > numOperators {
+			numOperators = len(oprStakes)
+		}
+		for opID := range oprStakes {
+			if _, ok := keyPairs[opID]; ok {
+				continue
+			}
+			keyPair, err := core.GenRandomBlsKeys()
+			if err != nil {
+				return nil, err
+			}
+			keyPairs[opID] = keyPair
 		}
-		keyPairs[ind] = keyPair
 	}
 
 	return &ChainDataMock{
-		NumOperators: numOperators,
 		KeyPairs:     keyPairs,
+		NumOperators: uint8(numOperators),
 		Stakes:       stakes,
 	}, nil
-
 }
 
-func MakeChainDataMock(numOperators core.OperatorIndex) (*ChainDataMock, error) {
-
-	stakes := make([]int, numOperators)
-	for ind := core.OperatorIndex(0); ind < numOperators; ind++ {
-		stakes[ind] = int(ind + 1)
+// MakeChainDataMock creates a ChainDataMock with a given number of operators per quorum
+// For example, given
+//
+//	numOperatorsPerQuorum = map[core.QuorumID]int{
+//		 0: 2,
+//		 1: 3,
+//	}
+//
+// It will create a ChainDataMock with 2 operators in quorum 0 and 3 operators in quorum 1
+// with stakes distributed as
+//
+//	map[core.QuorumID]map[core.OperatorID]int{
+//	  0: {
+//		   core.OperatorID{0}: 1,
+//		   core.OperatorID{1}: 2,
+//	  },
+//	  1: {
+//		   core.OperatorID{0}: 1,
+//		   core.OperatorID{1}: 2,
+//		   core.OperatorID{2}: 3,
+//	  },
+//	}
+func MakeChainDataMock(numOperatorsPerQuorum map[core.QuorumID]int) (*ChainDataMock, error) {
+	stakes := make(map[core.QuorumID]map[core.OperatorID]int)
+	for quorumID, numOpr := range numOperatorsPerQuorum {
+		stakes[quorumID] = make(map[core.OperatorID]int)
+		for i := 0; i < numOpr; i++ {
+			id := MakeOperatorId(i)
+			stakes[quorumID][id] = int(i + 1)
+		}
 	}
-	return NewChainDataMock(stakes)
 
+	return NewChainDataMock(stakes)
 }
 
 func (d *ChainDataMock) GetTotalOperatorState(ctx context.Context, blockNumber uint) *PrivateOperatorState {
 	return d.GetTotalOperatorStateWithQuorums(ctx, blockNumber, []core.QuorumID{})
 }
 
-func (d *ChainDataMock) GetTotalOperatorStateWithQuorums(ctx context.Context, blockNumber uint, quorums []core.QuorumID) *PrivateOperatorState {
+func (d *ChainDataMock) GetTotalOperatorStateWithQuorums(ctx context.Context, blockNumber uint, filterQuorums []core.QuorumID) *PrivateOperatorState {
+	quorums := filterQuorums
+	if len(quorums) == 0 {
+		for quorumID := range d.Stakes {
+			quorums = append(quorums, quorumID)
+		}
+	}
+
 	indexedOperators := make(map[core.OperatorID]*core.IndexedOperatorInfo, d.NumOperators)
 	storedOperators := make(map[core.OperatorID]*core.OperatorInfo)
 	privateOperators := make(map[core.OperatorID]*PrivateOperatorInfo, d.NumOperators)
 
-	var aggPubKey *core.G1Point
-
-	quorumStake := 0
-
-	for ind := core.OperatorIndex(0); ind < d.NumOperators; ind++ {
-		if ind == 0 {
-			key := d.KeyPairs[ind].GetPubKeyG1()
-			aggPubKey = key.Deserialize(key.Serialize())
-		} else {
-			aggPubKey.Add(d.KeyPairs[ind].GetPubKeyG1())
+	aggPubKeys := make(map[core.QuorumID]*core.G1Point)
+	for i := 0; i < int(d.NumOperators); i++ {
```

Comment:
I think this isn't working correctly, since it's using global indices across all of the quorums and using a single stake across all of the quorums. 

I'd suggest something like. 
- A loop over NumOperators to set the global parameters for each operator by id (indexed and private)
- A loop over each of the quorums in d.Stakes, in an order sorted by the operator id, to get the stake amount and index for the stored variables.